lib/components/screens/lists/members/bloc/list_member_event.dart

```diff
@@ -0,0 +1,90 @@
+part of 'list_member_bloc.dart';
+
+abstract class ListMemberEvent {
+  const ListMemberEvent();
+
+  Stream<ListMemberState> applyAsync({
+    required ListMemberState currentState,
+    required ListMemberBloc bloc,
+  });
+}
+
+class ShowListMembers extends ListMemberEvent with HarpyLogger {
+  const ShowListMembers();
+
+  @override
+  Stream<ListMemberState> applyAsync({
+    required ListMemberState currentState,
+    required ListMemberBloc bloc,
+  }) async* {
+    log.fine('load list members');
+
+    yield const ListMemberInitialLoading();
+
+    final paginatedUsers = await bloc.listsService
+        .members(listId: bloc.list.idStr)
+        .handleError(twitterApiErrorHandler);
+
+    if (paginatedUsers != null) {
+      final members =
+          paginatedUsers.users!.map((user) => UserData.fromUser(user)).toList();
+
+      if (members.isNotEmpty) {
+        yield ListMembersResult(
+          members: members,
+          membersCursor: paginatedUsers.nextCursorStr,
+        );
+      } else {
+        yield const NoListMembersResult();
+      }
+    } else {
+      yield const ListMemberFailure();
+    }
+  }
+}
+
+class LoadMoreMembers extends ListMemberEvent with HarpyLogger {
+  const LoadMoreMembers();
+
+  @override
+  Stream<ListMemberState> applyAsync({
+    required ListMemberState currentState,
+    required ListMemberBloc bloc,
+  }) async* {
+    if (currentState is ListMembersResult && currentState.hasMoreData) {
+      log.fine('load more members');
+
+      yield MembersLoadingMore(
+        members: currentState.members,
+        membersCursor: currentState.membersCursor,
+      );
+
+      final paginatedMembers = await bloc.listsService
+          .members(listId: bloc.list.idStr, cursor: currentState.membersCursor)
+          .handleError(twitterApiErrorHandler);
+
+      if (paginatedMembers != null) {
+        final members = List<UserData>.of(
+          currentState.members.followedBy(
+            paginatedMembers.users!
+                .map((user) => UserData.fromUser(user))
+                .toList(),
+          ),
+        );
+
+        yield ListMembersResult(
+          members: members,
+          membersCursor: paginatedMembers.nextCursorStr,
+        );
+
+        bloc.requestMoreCompleter.complete();
+        bloc.requestMoreCompleter = Completer<void>();
```

Comment:
This shouldn't be in the if since the ui always awaits the future and if it never completes in a fail state the loading indicator never stops showing